Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,189 @@
+                {
+                    SoundSystem.Play(Filter.Pvs(component.HostedGuardian), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.20f));
+                    EntityManager.QueueDeleteEntity(component.HostedGuardian);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
+        }
+
+        /// <summary>
+        /// Handles guardian reciving damage and splitting it with the host according to his defence percent
+        /// </summary>
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            if (EntityManager.TryGetComponent<DamageableComponent>(uid, out DamageableComponent? damageable))
+            {
+                EntityManager.TryGetComponent<DamageableComponent>(component.Host, out DamageableComponent? hostdamage);
+                if (hostdamage != null)
+                {
+                    if (args.DamageDelta != null)
+                    {
+                        EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamageShare));
+                        _popupSystem.PopupCoordinates(Loc.GetString("guardian-entity-taking-damage"), hostdamage.Owner.Transform.Coordinates, Filter.Entities(hostdamage.OwnerUid));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Triggers while trying to examine an activator to see if it's used
+        /// </summary>
+        private void OnActivatorExamined(EntityUid uid, GuardianCreatorComponent component, ExaminedEvent args)
+        {
+           if (component.Used == true)
+           {
+               string usedstring = Loc.GetString("guardian-activator-empty-invalid-creaton");
+               args.Message.AddMarkup("\n" + $"[color=#ba1919]{usedstring}[/color]");
+           }
+        }
+
+        /// <summary>
+        /// Called every time the host moves, to make sure the distance between the host and the guardian isn't too far
+        /// </summary>
+        private void OnGuardianHostMove(EntityUid uid, GuardianHostComponent component, ref MoveEvent args)
+        {
+            if (EntityManager.TryGetEntity(component.HostedGuardian, out IEntity? guard))
+            {
+                if  (guard.TryGetComponent<GuardianComponent>(out GuardianComponent? guardcomp))
+                {
+                    if (guardcomp.GuardianLoose == true)
+                    {
+                        //Compares the distance to allowed distance, otherwise forces a recall action from the host
+                        if (!guardcomp.Owner.Transform.Coordinates.InRange(EntityManager, component.Owner.Transform.Coordinates, guardcomp.DistanceAllowed))
+                        {
+                            OnGuardianManifestAction(guardcomp.OwnerUid, uid);
+                        }
+                    }
+                }        
+            }
```

Comment:
ECS, don't TryGetEntity and don't use the Transform property. You also don't need to do == true.